SigV4 client builder panics

Hello,

I am trying to implement the sigv4 package and construct a new sigv4 client. My code looks like this:

```
	// If using Amazon Managed Prometheus Address, create sigv4 roundtripper
	if strings.Contains(address, "aps-workspaces") {
		var cfg sigv4.SigV4Config
		var next http.RoundTripper
		sigv4RoundTripper, err := sigv4.NewSigV4RoundTripper(&cfg, next)
		if err != nil {
			log.Errorf("Error creating SigV4 RoundTripper: %v", err)
		}
		pc.RoundTripper = sigv4RoundTripper
	}
```

I am getting a panic when running this and i'm not sure why:

```
2023-03-30T16:27:57.664424-04:00 ??? Log level set to info
2023-03-30T16:27:57.664611-04:00 INF Starting cost-model version dev (HEAD)
2023-03-30T16:27:57.664693-04:00 INF Prometheus/Thanos Client Max Concurrency set to 5
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x104dd3fa0]

goroutine 54 [running]:
bytes.(*Buffer).ReadFrom(0x14000e81590, {0x0, 0x0})
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/bytes/buffer.go:202 +0x70
io.copyBuffer({0x1071e1200, 0x14000e81590}, {0x0, 0x0}, {0x0, 0x0, 0x0})
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/io/io.go:413 +0x108
io.Copy(...)
        /opt/homebrew/Cellar/go/1.19.5/libexec/src/io/io.go:386
github.com/prometheus/common/sigv4.(*sigV4RoundTripper).RoundTrip(0x14000dbb540, 0x14000635e00)
net/http.(*Client).Do(...)
```

I was thinking it's possible because I am implementing an empty ```sigv4.SigV4Config``` but I believe I should be able to implement this empty and it should just take the default AWS provider chain. Any idea why I am getting a panic here?

--- a/sigv4/sigv4.go
+++ b/sigv4/sigv4.go
@@ -110,11 +110,14 @@ func (rt *sigV4RoundTripper) RoundTrip(req *http.Request) (*http.Response, error
 		buf.Reset()
 		rt.pool.Put(buf)
 	}()
-	if _, err := io.Copy(buf, req.Body); err != nil {
-		return nil, err
+
+	if req.Body != nil {
+		if _, err := io.Copy(buf, req.Body); err != nil {
+			return nil, err
+		}
+		// Close the original body since we don't need it anymore.
+		_ = req.Body.Close()
 	}
-	// Close the original body since we don't need it anymore.
-	_ = req.Body.Close()
 
 	// Ensure our seeker is back at the start of the buffer once we return.
 	var seeker io.ReadSeeker = bytes.NewReader(buf.Bytes())

--- a/sigv4/sigv4_test.go
+++ b/sigv4/sigv4_test.go
@@ -107,4 +107,11 @@ func TestSigV4RoundTripper(t *testing.T) {
 
 		require.Equal(t, "/test/test", gotReq.URL.Path)
 	})
+
+	t.Run("No body", func(t *testing.T) {
+		req, err := http.NewRequest(http.MethodGet, "https://example.com/test/test", nil)
+		require.NoError(t, err)
+		_, err = cli.Do(req)
+		require.NoError(t, err)
+	})
 }

